### DISCUSSION TITLE
[v5] How to approach Migrations of a new adonis app, with an existing production database?

### DISCUSSION THREAD

**Discussion by ivan-prats** [adonisjs/core#2250] [Help] [answered]
A little bit of context:
- we currently have an app with another framework (Sails.js)
- it uses a postgres database in production, filled with customer data 
- we want to migrate to Adonis, re-doing the app from 0

We obviously want to keep all the data in our production DB. But how should we do it? 

I see the following possible options:
1. Take a "new database" approach. Doing "table.create" migrations, and then migrating the data from the current production to the new 
2. Take a "use the current database" approach. Using the current databse with the new Adonis app.

Anyone with experience on either of the approaches? I don't even know how to approach the second one for testing, not having a "table.create" migration to begin with.

**Comment by ammezie** [comment, marked as answer]
First, are you keeping the tables and columns as is?

Assuming yes:

If you care of migrations, then the first option will be your best bet. Create the migrations for each tables, then migrate the data over using a script or manually. Otherwise, simply use the existing database directly.

**Reply by ivan-prats** [reply]
If I were to keep the columns how they are: how should I approach testing? I do need to migrate and revert the database to be able to test properly

**Reply by thetutlage** [reply]
I agree, for your test database, you need migrations to create tables. What you can do in this situation is create migrations for all the existing tables. Instead of using `table.create` directly, you can wrap it inside the `hasTable` conditional
   
```ts
  if (!(await this.schema.hasTable(this.tableName))) {
    this.schema.createTable(this.tableName, (table) => {})
  }
```

This way running these migrations against the existing database will not result in any error and will also create the migrations for tests.

Also, I recommend testing all your migrations locally first, before running them against the prod DB

**Reply by ivan-prats** [reply]
Thanks @thetutlage ! That is definetely what i was looking for in case i decide to go for that option.
